packages/phone-number-privacy/signer/test/integration/domain.test.ts

```diff
@@ -0,0 +1,708 @@
+
+      const res = await request(app).post(SignerEndpoint.DOMAIN_QUOTA_STATUS).send(badRequest)
+
+      expect(res.status).toBe(401)
+      expect(res.body).toMatchObject<DomainQuotaStatusResponse>({
+        success: false,
+        version: res.body.version,
+        error: WarningMessage.UNAUTHENTICATED_USER,
+      })
+    })
+
+    it('Should respond with 503 on disabled api', async () => {
+      _config.api.domains.enabled = false
+      const appWithApiDisabled = startSigner(_config, db, keyProvider)
+
+      const req = await quotaRequest()
+
+      const res = await request(appWithApiDisabled)
+        .post(SignerEndpoint.DOMAIN_QUOTA_STATUS)
+        .send(req)
+
+      expect(res.status).toBe(503)
+      expect(res.body).toMatchObject<DomainQuotaStatusResponse>({
+        success: false,
+        version: res.body.version,
+        error: WarningMessage.API_UNAVAILABLE,
+      })
+    })
+  })
+
+  describe(`${SignerEndpoint.DOMAIN_SIGN}`, () => {
+    it('Should respond with 200 on valid request', async () => {
+      const [req, _] = await signatureRequest()
+
+      const res = await request(app).post(SignerEndpoint.DOMAIN_SIGN).send(req)
+
+      expect(res.status).toBe(200)
+      expect(res.body).toMatchObject<DomainRestrictedSignatureResponse>({
+        success: true,
+        version: res.body.version,
+        signature: res.body.signature,
+        status: {
+          disabled: false,
+          counter: 1,
+          timer: res.body.status.timer,
+          now: res.body.status.now,
+        },
+      })
+    })
+
+    it('Should respond with 200 on valid request with key version header', async () => {
+      const [req, _] = await signatureRequest()
+
+      const res = await request(app)
+        .post(SignerEndpoint.DOMAIN_SIGN)
+        .set('keyVersion', '1')
+        .send(req)
+
+      expect(res.status).toBe(200)
+      expect(res.body).toMatchObject<DomainRestrictedSignatureResponse>({
+        success: true,
+        version: res.body.version,
+        signature: res.body.signature,
+        status: {
+          disabled: false,
+          counter: 1,
+          timer: res.body.status.timer,
+          now: res.body.status.now,
+        },
+      })
+    })
+
+    it('Should respond with 200 on repeated valid requests', async () => {
+      const req1 = (await signatureRequest())[0]
+
+      const res1 = await request(app)
+        .post(SignerEndpoint.DOMAIN_SIGN)
+        .set('keyVersion', '1')
+        .send(req1)
+
+      console.log(res1.body)
+
+      expect(res1.status).toBe(200)
+      expect(res1.body).toMatchObject<DomainRestrictedSignatureResponse>({
+        success: true,
+        version: res1.body.version,
+        signature: res1.body.signature,
+        status: {
+          disabled: false,
+          counter: 1,
+          timer: res1.body.status.timer,
+          now: res1.body.status.now,
+        },
+      })
+
+      // submit identical request with nonce set to 1
```

Comment:
is req2 supposed to be actually identical to req1 except for the nonce (and sig over request ofc)? It's currently also different in the `sessionID` & `blindedMessage` -- we should either update the comment or change the test depending on what expected behavior here is